RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.h

```diff
@@ -0,0 +1,62 @@
+#ifndef RecoParticleFlow_PFClusterProducer_PFClusterProducerAlpakaKernel_h
+#define RecoParticleFlow_PFClusterProducer_PFClusterProducerAlpakaKernel_h
+
+#include "DataFormats/ParticleFlowReco/interface/alpaka/PFRecHitDeviceCollection.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitHostCollection.h"
+#include "DataFormats/ParticleFlowReco/interface/alpaka/PFClusterDeviceCollection.h"
+#include "DataFormats/ParticleFlowReco/interface/alpaka/PFRecHitFractionDeviceCollection.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusterParamsDeviceCollection.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusteringVarsDeviceCollection.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusteringEdgeVarsDeviceCollection.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  namespace reco::pfClustering {
+    struct Position4 {
```

Comment:
When using `Position4` instead of `float4`, do you know if the CUDA compiler generates optimised read/store instructions, as discussed in [CUDA Pro Tip: Increase Performance with Vectorized Memory Access](https://developer.nvidia.com/blog/cuda-pro-tip-increase-performance-with-vectorized-memory-access/) ?

Should Alpaka provide a portable implementation of `float4`, `float3`, `int4`, _etc_ ?